spec/rails_helper.rb

```diff
@@ -215,8 +215,14 @@ def seed_base_data_for_tests
   end
 
   config.before(:each) do
-    # Defined shared @ global variables used throughout the test suite.
-    define_global_variables if RSpec.current_example.metadata[:seed_items] != false
+    # This is a workaround to move towards not seeding the database at the suite level.
+    # ENV flag is used to speed up local development.
+    if RSpec.current_example.metadata[:seed_db] == false
+      DatabaseCleaner.clean_with(:truncation) unless ENV["SKIP_SEED"] == "true"
```

Comment:
I don't think the truncation clean is necessary. Really all we need to do is _not_ seed the database at the start. Wondering if it's faster to just create the new data when necessary than to truncate on every test.

One thing that might be worth investigating is using `before(:all)`. This runs outside the transaction. We would assume that an entire test block either _is_ or _is not_ using `seed_items`. In this case you could put the `seed_items` on the block level, not the test level, and truncate the tables only if it's true. (Over time, we could assume it's true and only truncate + _insert_ the data if it's false.)

Reply:
After a bit on investigation.

### Make Seeding Items in Each Test the Default

```rb
unless RSpec.current_example.metadata[:seed_db] == false
  seed_base_data_for_tests  # Defined shared @ global variables used throughout the test suite.
  define_global_variables
end
```

Result:
Not seeding the db => seeding on every test results in all models going from 30s locally to 2:44 min over 655 tests. Keep in mind models are by far the worst case scenario and I already have PRs ready to fix all the models. That said, CI time seems to have only increased by about 2min. The approach seems reasonable to me.

### Seed with before(:all)

  ```rb
config.before(:all, :seed_db) do
  seed_base_data_for_tests
  define_global_variables
end

config.after(:all, :seed_db) do
  DatabaseCleaner.clean_with(:truncation, except: %w[ar_internal_metadata])
end
```

Result:
Not seeding db => Seeding before all specs in a file then tearing down after results in all models going from 30s locally to 1 min over 655 tests. Its a good bit faster than the seeding on every test approach but I will have to add the `seed_db: true` to every single spec.

Reply:
OK - let's fix the specs that do that with the global variables first.

Can we do it so that rather than putting `seed_db: true` on each spec, we have something like `skip_seed_db: true`and only put that on the specs we _don't_ want to seed?

Another thing to think about - can we keep track of the _previous_ example group? And if both this and the previous group both had `skip_db: false` (or `seed_db: true`) then we don't have to do anything around truncation since the data should still be accurate?

Reply:
> OK - let's fix the specs that do that with the global variables first.

Sounds like a plan. 

> Can we do it so that rather than putting seed_db: true on each spec, we have something like skip_seed_db: true and only put that on the specs we don't want to seed?

Maybe you have some suggestions on how to get that to work? I might be missing something super obvious. I can't check the metadata for a tag within a `before(:all)` or `before(:context)`. `around(:all)` and `around(:context)` aren't supported and just drop down to `around(:each)`. 

It seems like the only approach is a tag in the config block `config.before(:all, seed_db: true) do` but I am just totally lost on how to make it run on the absence of a tag.

> Another thing to think about - can we keep track of the previous example group? And if both this and the previous group both had skip_db: false (or seed_db: true) then we don't have to do anything around truncation since the data should still be accurate?

I'll investigate.

Reply:
@elasticspoon it looks like you can do this actually! rspec/rspec-core#2728 Check the last comment.

Reply:
Nice! Made the changes.